### DISCUSSION TITLE
Disable all rules of a plugin into specific files

### DISCUSSION THREAD

**Discussion by quirino-cargill** [eslint/eslint#15216] [Show and tell]
Found one issue where was discussed to disable all the rules of a eslint plugin. 
Had the same need and ended up doing that in JS just importing all the rules of the plugin from its module and reducing the keys with rules exported, and adding `off` as value. 

It looks to work fine when linting through cli (IDE may not pick that up though) 

Example of just an eslintrc js config file inside sub path where we wanted these rules off from the unicorn plugin: 

```
const eslintPluginToDisable = require('eslint-plugin-unicorn');

const disabledRules = Object.keys(eslintPluginToDisable.configs.recommended.rules).reduce((acc, val) => {
  return { ...acc, [val]: 'off' };
}, {});

module.exports = {
  rules: {
    ...disabledRules,
  },
};

-  

```

Hope will be helpful and guys you will enjoy coding without frustration of stricting coding styling rules.

**Comment by JuliusKoronci** [comment]
The only downside of this approach is when the plugin also introduces configurations for other plugins. E.g. if your app extends multiple plugins and the one your are disabling like this introduces rules for them, it will not be reset but disabled. A better approach would be actually filtering out the plugin altogether from the plugins in a config which extends your configuration
